File: src/my_project_typescript/easy_problems_typescript/from51To100/removeTheOutermostParenthesis.ts
function removeOuterParentheses(s: string): string {

    let leftCount: number = 0
    let rightCount: number = 0
    let count: number = 0
    let answer: string = ""

    for (let character of s){
        if (character === "("){
            leftCount += 1
        }
        else if (character === ")"){
            rightCount += 1
        }
        count += 1
        
        if (leftCount === rightCount ){
            answer += s.slice((count - 2*leftCount + 1), count-1)
            leftCount = 0
            rightCount = 0
        }
    }

    return answer 

};


console.log(removeOuterParentheses("(()())(())"))

console.log(removeOuterParentheses("(()())(())(()(()))"))


console.log(removeOuterParentheses("()()"))

File: src/my_project/easy_problems/from151To200/remove_the_outermost_parenthesis.py
from typing import List 

class Solution:
    def removeOuterParentheses(self, s: str) -> str:

        left_count: int = 0
        right_count: int = 0
        count = 0
        answer = ""

        for character in s: 
            if character == "(":
                left_count += 1
            elif character == ")":
                right_count += 1
            
            count += 1
            if left_count == right_count:
                answer += s[(count - 2*left_count + 1): count-1]
                left_count = 0
                right_count = 0
                

            



        return answer

solution = Solution()

print(solution.removeOuterParentheses(s = "(()())(())"))

print(solution.removeOuterParentheses(s = "(()())(())(()(()))"))

print(solution.removeOuterParentheses(s = "()()"))



print("abcdef"[1:3])